### PR TITLE
share upload dive-share code (preparing for mobile)

### PR DESCRIPTION
**Pull request by janiversen**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

moving the actual upload code from desktop-widgets to core.

This PR only rearranges the code to make it useable for the mobile version.

A new PR will make the mobile integration.

### CODE REVIEW

File: core/uploadDiveShare.cpp
Comment (dirkhh) [inline core/uploadDiveShare.cpp:60]: silly bike-shedding comment about the name. Again, no need to refactor for this, can always simply be renamed later.

File: desktop-widgets/diveshareexportdialog.cpp
Comment (dirkhh) [inline desktop-widgets/diveshareexportdialog.cpp:117]: this follows the pattern of splitting a code move across two commits.
I understand your reason for it. maybe I shouldn't even comment on it.
this is simply a preference on my part to have that in the same commit...

**Review by dirkhh — APPROVED** [review]
I haven't tested the code - I'm sure you did. But from a review perspective I think this is good to go.
I assume there's no reason to wait for the mobile code, correct?

**Comment by dirkhh** [comment]
Thank you for following my suggestion from yesterday to split this into a smaller PR just factoring out the code. This makes the review so much easier.
I should have time later this morning to do a quick sanity test on this one, but unless you tell me otherwise, I expect to merge this today.

**Comment by dirkhh** [comment]
OK, tested and works exactly as before (which doesn't mean that it works well... for example in my test it exported the end pressure of a tank, but not the start pressure - but that is the case with both your code and master, so it has nothing to do with this PR)
Ready to merge from my perspective.

**Comment by janiversen** [comment]
No reason to wait for the other, it can be merged. Once merged I need to rebase the other and do a push -f, but I will not make changes

**Pull request merged**
